log-group-events-lambda/handler/sf_ops.go

```diff
@@ -0,0 +1,169 @@
+package handler
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs/cloudwatchlogsiface"
+	"github.com/hashicorp/go-multierror"
+	"github.com/logzio/firehose-logs/common"
+	"os"
+	"sync"
+	"time"
+)
+
+type CloudWatchLogsClient struct {
+	Client cloudwatchlogsiface.CloudWatchLogsAPI
```

Comment:
Whats is the advantage of having the struct instead of using `cloudwatchlogsiface.CloudWatchLogsAPI`

Reply:
It's needed for the mock tests, I added to the struct the mutex that we use in the go routine too now